### DISCUSSION TITLE
Not very intuitive way of work when generating WordWise on EPUB

### DISCUSSION THREAD

**Discussion by Phobooky** [xxyzz/WordDumb#61] [Ideas]
I'd like share with you a way of work that I think it is not very intuitive when generating files with the plugin.
I'm talking when you're generating files in non English books.
If you try to generate WordWise y the preferred format file exists, it works if a different way respect than the preferred format file does not exists.
Example: Let me suppose I want to generate Word Wise on EPUB and my  preferred format file is KFX:
If I try to generate  WordWise on EPUB book and the KFX book does not exist, the plugin generate the EPUB book with WordWise perfectly.
But, If I try to generate WordWise on EPUB and the KFX book exists, the plugin generate the following message: "Non-English book: For Kindle format books, Word Wise is only available in books in English.". 
In this case, to solve it, I must to delete the KFX book or to change my preferred format file to EPUB to solve it. Not very intuitive.
In general, if it exist any book in Calibre with a format more above in the preferred format file list than EPUB, the plugin gives the error and does not generate WordWise on EPUB.
It is very difficult for newies to guess this.
Perhaps it would be easier to have a button only for WordWise on EPUB to generate it independiently from the order in preferred format file list and independiently of the books in different formats I have already generated. Another solution could be to change the message giving the apropiate information.

**Comment by xxyzz** [comment]
How about open a dialog for users to choose format like calibre's choose format to edit dialog? And add a new option in the plugin settings to let the plugin selects preferred format automatically.

**Reply by Phobooky** [reply]
I think this is a good way to solve the issue. 👍

**Reply by xxyzz** [reply]
I have added the feature to the master branch, please check out the GitHub Actions.

**Comment by Phobooky** [comment]
Perfect solution. I've test it and works fine, but I've only found two problems.

1. If you check "Create files for all available formats"
2. You have not EPUB as first format in the list (in this case, I have KFX first) 
3. You have books in several formats (as KFX, AZW3 and EPUB)


When you create Word Wise you receive the message:

**Non-English book: For Kindle format books, Word Wise is only available in books in English.**

I think you are supposed to create Word Wise on EPUB although a message could say that only WordWise on EPUB will be generated.
In other case, the user should unckeck "Create files for all available formats" or change the priority. Not very intuitive, as before this change.

![image](https://user-images.githubusercontent.com/24461946/183245141-cd598b6f-0140-4fed-ab67-ab2ce850332d.png)

![image](https://user-images.githubusercontent.com/24461946/183245152-d077cea8-c202-43e2-b265-fc7bddaf8d45.png)

The other thing:
The other thing: The plugin lets you this configuration. I think it should be to have one option selected in a mandatory basis. None of them selected has not any sense.

![image](https://user-images.githubusercontent.com/24461946/183245411-c7c4b88c-7d22-4dd3-897f-6174c36f093d.png)

**Reply by xxyzz** [reply]
I add that message so people won't ask why the plugin does nothing for unsupported language/format and think it's a bug. By default, "Choose format manually" is enabled and "Create files for all available formats" is disabled.

That warning dialog is just a warning, you can ignore it, X-Ray file is still created.

**Reply by Phobooky** [reply]
Ok. I understand you way of thinking. Thanks. Be aware with unselected both of options.

**Reply by xxyzz** [reply]
Uncheck both options is fine: move your mostly used format to the top and let the plugin automatically choose that format.

"Create files for all available formats" is for people use both Kindle and non-Kindle devices and want to create files for  both devices, most people won't need this option.

**Reply by Phobooky** [reply]
Ok. I understand. Thanks.
